test/Microsoft.AspNet.Mvc.TagHelpers.Test/OptionTagHelperTest.cs

```diff
@@ -172,11 +427,10 @@ public static IEnumerable<object[]> DoesNotUseViewContextDataSet
             await tagHelper.ProcessAsync(tagHelperContext, output);
 
             // Assert
-            Assert.Equal(
-                expectedOutput,
-                output.GenerateStartTag() +
-                (output.GenerateContent() as TagHelperContent).GetContent() +
-                output.GenerateEndTag());
+            Assert.Equal(expectedTagHelperOutput.TagName, output.TagName);
+            Assert.Equal(expectedTagHelperOutput.Content.GetContent(), output.Content.GetContent());
+            Assert.Equal(expectedTagHelperOutput.Attributes.Count, output.Attributes.Count);
+            Assert.True(expectedTagHelperOutput.Attributes.All(e => output.Attributes.Contains(e)));
```

Comment:
`Assert.True` makes this hard to debug.  probably better to do a `foreach` over `expectedTagHelperOutput.Attributes` and `Assert.Contains()` in the loop body.

could also `Assert.Equal()` the two enumerations since order is important.  see if that works -- it may not due to `Dictionary` vagaries.